application/common/models/MfaBackupcode.php

```diff
@@ -26,12 +26,14 @@ public function rules(): array
      * Check if given value exists, if so delete and return true, else false
      * @param int $mfaId
      * @param int $code
+     * @param bool $sendEmail optional (default=True) Whether to send an email (False is used by tests)
      * @return bool
      * @throws ServerErrorHttpException
      */
-    public static function validateAndRemove(int $mfaId, int $code): bool
+    public static function validateAndRemove(int $mfaId, int $code, $sendEmail=True): bool
```

Comment:
It looks like we can remove the need for this `$sendEmail` parameter by simply changing the relevant test. If we simply have the `When to send refresh backup codes emails` test remove records of any emails that have been sent as the last step before the `When` (instead of earlier in the list of `Given`s), then the test works as desired and we don't need to add this parameter.

Reply:
Done